odoo/fields.py

```diff
@@ -345,19 +345,22 @@ def _get_attrs(self, model, name):
 
     def _setup_attrs(self, model, name):
         """ Initialize the field parameter attributes. """
+        # validate extra arguments
+        for key in self.args:
```

Comment:
The field parameters are in `attrs` below, not `self.args`.

Actually, `self.args` will be empty most of the time here.  Consider a model definition like
```python
class Stuff(models.Model):
    _name = 'stuff'
    foo = fields.Char(stuff=42)    # Stuff.foo.args = {'stuff': 42}
```
The class put in the registry will actually inherit from `Stuff` and look like
```python
class ModelClass(Stuff, ...):
    _name = 'stuff'
    foo = fields.Char()    # ModelClass.foo.args = {}
```
The method `_setup_attrs` is actually invoked on `ModelClass.foo`, which introspects its class definition and determine `attrs` from `Stuff.foo.args` and `ModelClass.foo.args`.  The latter is non-empty only for automatic and custom fields.

Simply try the branch without the fixes, and you will see that it does not do what it is supposed to.